contracts/contracts/vault/VaultCore.sol

```diff
@@ -158,6 +162,23 @@ contract VaultCore is VaultStorage {
     function _redeem(uint256 _amount, uint256 _minimumUnitAmount) internal {
         require(_amount > 0, "Amount must be greater than 0");
 
+        uint256 _totalSupply = oUSD.totalSupply();
+        uint256 _backingValue = _totalValue();
+
+        if (maxSupplyDiff > 0) {
+            // Allow a max difference of maxSupplyDiff% between
+            // backing assets value and OUSD total supply
+            require(
+                (
+                    _totalSupply > _backingValue
+                        ? _totalSupply.div(_backingValue)
+                        : _backingValue.div(_totalSupply)
+                )
+                    .mul(100 ether) <= maxSupplyDiff,
```

Comment:
Can we use StableMath here? Most other units we store are expressed in 18 decimal.

maxSupplyDiff should be expressed like 2e17 for 20%.